Add TAS5825M

aurel32 opened the pull request:
Add the TAS5825M chip from TI which is 38-W Stereo Class-D Audio Amplifier in QFN32 package.

http://www.ti.com/lit/ds/symlink/tas5825m.pdf

![tas5825m](https://user-images.githubusercontent.com/7688913/56603260-ad4bc280-65ff-11e9-8e84-5467a26ae9cf.png)

The library checker complains about the following violations:
```
Checking symbol 'TAS5825M':
  Violating S4.2
    Pins should be arranged by function
    Positive power pins should be placed at top of symbol
     - Pin AVDD (19) @ (-600,600)
checking ruleS4.3
checking ruleS4.4
  Violating S4.4
    Pin electrical type should match pin function
    Power pins should be of type POWER INPUT or POWER OUTPUT
     - Pin AVDD (19) @ (-600,600) is of type PASSIVE

```

The VR_DIG, GVDD and AVDD are decoupling pins from the internal regulators. They are therefore not power pins. However AVDD is wrongly detected by the check scripts as a power pin, and this causes a violation of rules S4.2 and S4.4


---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the symbol(s) you are contributing
- [x] An example screenshot image is very helpful
- [x] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
  - A new fitting footprint must be submitted if the library does not yet contain one.
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required
- [x] Give a reason behind any intentional library convention rule violation.

myfreescalewebpage commented:
Hi @aurel32 , thanks for contributing,

A few comments I have during my review:

- [ ] Name of the symbol should be `TAS5825MHB` to avoid futur conflict if the manufacturer add package options
- [ ] Package in the description is not correct, should be "VQFN-32"
- [ ] Can you align DVDD with DGND ?
- [ ] Name of pin 9 is `GPIO0`, name of pin 10 is `GPIO1` and name of pin 11 is `GPIO2`
- [ ] All BST* pins should be Passive to avoid ERC errors
- [ ] Name of pin 33 should be `EP`
- [ ] I think it will be better to move `VR_DIG`, `GVDD`and `AVDD` to the top of the symbol, but keep them Passive. Moreover, `GVDD` should be on the right.

![Capture](https://user-images.githubusercontent.com/16516760/57107423-3ebde180-6d30-11e9-9223-6ce34557a26c.PNG)


Cheers,
Joel

aurel32 commented:
Hi,

> A few comments I have during my review:

Thanks for the review. Here's a screenshot of the new version:
 
![tas5825mrhb](https://user-images.githubusercontent.com/7688913/57254070-a174ed00-7050-11e9-8c83-dc3b9e9d10cc.png)

* [x]  Name of the symbol should be `TAS5825MHB` to avoid futur conflict if the manufacturer add package options

I guessed you mean `TAS5825MRHB`, in that case it's fixed. Actually it would be very nice if the manufacturer can release a TQFP32 of this nice chip ;-)

* [x]  Package in the description is not correct, should be "VQFN-32"

Fixed, thanks.

* [x]  Can you align DVDD with DGND ?

Fixed.

* [x]  Name of pin 9 is `GPIO0`, name of pin 10 is `GPIO1` and name of pin 11 is `GPIO2`

Good catch, fixed.

* [x]  All BST* pins should be Passive to avoid ERC errors

Fixed.

* [x]  Name of pin 33 should be `EP`

Fixed.

* [x]  I think it will be better to move `VR_DIG`, `GVDD`and `AVDD` to the top of the symbol, but keep them Passive. Moreover, `GVDD` should be on the right.

Also fixed.

myfreescalewebpage commented:
@aurel32 thanks, no more comment on this symbol. Merging.
